src/ansys/pre_commit_hooks/tech_review.py

```diff
@@ -0,0 +1,734 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Module for checking if a repository is compliant with required files in the technical review."""
+import argparse
+from datetime import date as dt
+from enum import Enum
+from itertools import product
+import json
+import pathlib
+import re
+from tempfile import NamedTemporaryFile
+
+import git
+from jinja2 import Environment, FileSystemLoader
+import requests
+import semver
+import toml
+
+from ansys.pre_commit_hooks.add_license_headers import check_same_content
+
+HOOK_PATH = pathlib.Path(__file__).parent.resolve()
+"""Location of the pre-commit hook on your system."""
+LICENSES_JSON = HOOK_PATH / "assets" / "licenses.json"
+"""JSON file containing licenses information."""
+DEFAULT_AUTHOR_MAINT_NAME = "ANSYS, Inc."
+"""Default name of project authors and maintainers."""
+DEFAULT_AUTHOR_MAINT_EMAIL = "pyansys.core@ansys.com"
+"""Default email of project authors and maintainers."""
+DEFAULT_START_YEAR = dt.today().year
+"""Default start year of the repository."""
+DEFAULT_LICENSE = "MIT"
+"""Default license of the repository"""
+JSON_URL = "https://raw.githubusercontent.com/spdx/license-list-data/main/json/licenses.json"
+"""URL to retrieve list of license IDs and names."""
```

Comment:
Suggesting to add a blank line between these objects:

```suggestion
HOOK_PATH = pathlib.Path(__file__).parent.resolve()
"""Location of the pre-commit hook on your system."""

LICENSES_JSON = HOOK_PATH / "assets" / "licenses.json"
"""JSON file containing licenses information."""

DEFAULT_AUTHOR_MAINT_NAME = "ANSYS, Inc."
"""Default name of project authors and maintainers."""

DEFAULT_AUTHOR_MAINT_EMAIL = "pyansys.core@ansys.com"
"""Default email of project authors and maintainers."""

DEFAULT_START_YEAR = dt.today().year
"""Default start year of the repository."""

DEFAULT_LICENSE = "MIT"
"""Default license of the repository"""

JSON_URL = "https://raw.githubusercontent.com/spdx/license-list-data/main/json/licenses.json"
"""URL to retrieve list of license IDs and names."""

```